### DISCUSSION TITLE
Container query precedence

### DISCUSSION THREAD

**Discussion by akbar-idwise** [tailwindlabs/tailwindcss#11296] [Help]
If we have some styles applied generically on a div and we want to override them based on container size shouldn't the container query take precedence? right now it doesn't seem to work e.g

```
<div class="@container h-screen w-screen">
  <div class="invisible @md:visible">Only visible on container size md and larger</div>
</div>
```

**Comment by wongjn** [comment]
Seems to work fine in Tailwind Play: https://play.tailwindcss.com/WrZI1ph7VF. This would suggest a configuration problem in your project. Consider checking:

- You have the `@tailwindcss/container-queries` NPM package installed.
- You have the `@tailwindcss/container-queries` plugin added to the `plugins` array in the Tailwind configuration.
- Tailwind has run a compilation since you added the container query classes.
- The file that the code appears in is covered by the `content` file globs in the Tailwind configuration.
- If you have customized the `containers` theme key, that `md` is a valid value.

**Reply by akbar-idwise** [reply]
It seems to work only if i make it important otherwise it doesn't `@md:!visible`

**Reply by wongjn** [reply]
Seems to work fine in Tailwind Play: https://play.tailwindcss.com/WrZI1ph7VF without `!important`. This would suggest a configuration problem in your project. Consider checking the points as mentioned previously or provide a reproducible example in a [Tailwind Play](https://play.tailwindcss.com) or a remotely-hosted git repository.

**Reply by akbar-idwise** [reply]
@wongjn I have been able to dig into the issue and i think the issue is with another plugin.
So we are using [postcss-scopify](https://www.npmjs.com/package/postcss-scopify) to scope all the styles to a single element but the styles generated by [tailwindcss-container-queries](https://github.com/tailwindlabs/tailwindcss-container-queries) plugin are not scoped. Rest of the styles generated by tailwind are being scoped.

This is our postcss config:
```
module.exports = {
  extract: true,
  plugins: [
    tailwindcss("./tailwind.config.js"),
    autoprefixer({}),
    postcssImport(),
    postcssScopify("#main"),
    postcssMinify(),
  ],
};
```
